projects/storefrontapp/src/app/spartacus/features/cdc/cdc-config-initializer.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, inject } from '@angular/core';
+import { CdcConfig } from '@spartacus/cdc/root';
+import {
+  BaseSiteService,
+  ConfigInitializer,
+} from '@spartacus/core';
+import { Observable, lastValueFrom } from 'rxjs';
+import { map, take } from 'rxjs/operators';
+
+@Injectable({ providedIn: 'root' })
+export class CdcConfigInitializer implements ConfigInitializer {
+  readonly scopes = ['cdc'];
+  readonly configFactory = () => lastValueFrom(this.resolveConfig());
+  protected baseSiteService = inject(BaseSiteService);
+  /**
+   * Emits the Cdc config basing on the current base site data.
+   *
+   * Completes after emitting the value.
+   */
+  protected resolveConfig(): Observable<CdcConfig> {
```

Comment:
what about move these logic to /spartacus/integration-libs/cdc/root/config

Reply:
have been modified.